### PR TITLE
Solution and editor configuration cleanup

### PR DESCRIPTION
**Pull request by withinfocus**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Using Visual Studio proper there are some annoyances popping up and files changing, so fix configuration in a few places.

## Code changes

* **.editorconfig:** Utilize what the template has and more globally apply formatting settings, specifically ending with newlines.
* **Directory.Build.props:** Found an errant version insertion.
* **bitwarden-server.sln:** Solution items were out of date with some missing files and others not included, so added what we have in the root.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -5,14 +5,18 @@ root = true
...
 # Don't use tabs for indentation.
 [*]
-end_of_line = lf
 indent_style = space
+tab_width = 4
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+guidelines = 120
```
Comment (MGibson1) [inline .editorconfig:14]: ⚠️ fine to move, but please remove the duplicate at the bottom

**Review by MGibson1 — CHANGES_REQUESTED** [review]

File: .editorconfig [hunk 2]
```diff
@@ -5,14 +5,18 @@ root = true
 
 # Don't use tabs for indentation.
 [*]
-end_of_line = lf
 indent_style = space
+tab_width = 4
```
[the remark this answers is not in the document]
Comment (withinfocus) [reply .editorconfig:9]: Nice -- didn't see that.

File: .editorconfig (hunk 1 above)
Comment (withinfocus) [reply .editorconfig:14]: Nice -- didn't see that.

File: .editorconfig (hunk 2 above)
[the remark this answers is not in the document]
Comment (withinfocus) [reply .editorconfig:9]: I guess I saw it as a starting point and it's pretty common, but for this repo and the diverse set of languages used I can pull it.

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**